astropy/table/column.py

```diff
@@ -980,6 +981,12 @@ def _compare(self, other):
             if self.dtype.char == 'S':
                 other = self._encode_str(other)
 
+            if NUMPY_LT_1_15 and not NUMPY_LT_1_14_5 and (
```

Comment:
I think we should get rid of the ``not NUMPY_LT_1_14_5`` here as the issue is present in other versions, e.g. 1.14.4